eox_tagging/api/v1/filters.py

```diff
@@ -25,38 +27,26 @@ class Meta:  # pylint: disable=old-style-class
         model = Tag
         fields = ['key', 'created_at', 'activated_at', 'status', 'course_id', 'enrolled', 'enrollments', 'username']
 
-    def filter_course_id(self, queryset, name, value):  # pylint: disable=unused-argument
-        """Filter that returns the tags associated with course_id."""
-        if value:
-            try:
-                queryset = Tag.objects.find_all_tags_for(target_type="CourseOverview",
-                                                         target_id={"course_id": str(value)})
-            except Exception:  # pylint: disable=broad-except
-                return queryset.none()
-
-        return queryset
-
-    def filter_username(self, queryset, name, value):  # pylint: disable=unused-argument
-        """Filter that returns the tags associated with username."""
-        if value:
-            try:
-                queryset = Tag.objects.find_all_tags_for(target_type="user",
-                                                         target_id={"username": str(value)})
-            except Exception:  # pylint: disable=broad-except
-                return queryset.none()
-
-        return queryset
-
-    def filter_enrolled(self, queryset, name, value):  # pylint: disable=unused-argument
-        """Filter that returns tags in which the target is a course where the user is enrolled in."""
+    def filter_by_target_object(self, queryset, name, value):
+        """Filter that returns the tags associated with target."""
+        TARGET_TYPES = {
+            "course": "courseoverview",
+            "user": "user",
+            "enrollment": "courseenrollment"
+        }
+
+        TARGET_IDENTIFICATION = defaultdict(lambda: str(value))
+        TARGET_IDENTIFICATION["enrollment"] = {
+            "username": self.request.user.username,
+            "course_id": str(value)
+        }
         if value:
-            enrollment = {  # pylint: disable=broad-except
-                "username": self.request.user.username,
-                "course_id": str(value)
-            }
             try:
-                queryset = Tag.objects.find_all_tags_for(target_type="courseenrollment",
-                                                         target_id=enrollment)
+                filter_params = {
+                    "target_type": TARGET_TYPES.get(name),
+                    "target_id": TARGET_IDENTIFICATION.get(name)
```

Comment:
trailing comma, is this a dict ??

Reply:
yes, why?

Reply:
Because I do a small test and this returns None
![image](https://user-images.githubusercontent.com/36200299/86141968-47d34b80-bab8-11ea-8082-fe0821110899.png)

Reply:
It seems like I cannot access it using .get because I'll get always None for a not defined key. Let me fix that. Thank you

Reply:
Fixed. I don't really know what the error was, I changed it to a regular dict